### PR TITLE
Message interface clean and documentation

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/message_interface/response.rs [hunk 1]
```diff
@@ -23,138 +23,189 @@ use bee_rest_api::types::{
...
-    /// Prepared transaction data for signing
+    /// Response for:
+    /// - [`PrepareTransaction`](crate::message_interface::Message::PrepareTransaction)
     PreparedTransactionData(PreparedTransactionDataDto),
-    /// Signed transaction data for signing
+    /// Response for:
+    /// - [`SignTransaction`](crate::message_interface::Message::SignTransaction)
     SignedTransaction(PayloadDto),
-    /// returns the unsynced nodes.
-    #[cfg(not(target_family = "wasm"))]
```
Comment (Thoralf-M) [inline src/message_interface/response.rs:68]: This still should stay?

Comment (thibault-martinez) [reply src/message_interface/response.rs:68]: Good catch

File: src/message_interface/message.rs [hunk 2]
```diff
@@ -1,25 +1,512 @@
...
+    GenerateBlock {
+        /// Secret manager
+        #[serde(rename = "secretManager")]
+        secret_manager: Option<SecretManagerDto>,
+        /// Options
+        options: Option<GenerateBlockOptions>,
+    },
+    /// Get a node candidate from the synced node pool.
+    GetNode,
+    // /// Gets the miner to use based on the PoW setting
```
Comment (Thoralf-M) [inline src/message_interface/message.rs]:
```suggestion
    /// Gets the miner to use based on the PoW setting
```

File: src/message_interface/message.rs [hunk 3]
```diff
@@ -1,25 +1,512 @@
...
+    // GetPoWProvider,
```
Comment (Thoralf-M) [inline src/message_interface/message.rs]: :angry:

Comment (thibault-martinez) [reply src/message_interface/message.rs]: 😠

**Review by Thoralf-M — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**